fabrics.c

```diff
@@ -937,6 +937,20 @@ int nvmf_discover(const char *desc, int argc, char **argv, bool connect)
 		}
 	}
 
+	/*
+	 * When --host-traddr/--host-iface are not specified on the command
+	 * line, use the discovery controller's (c) host-traddr/host-iface for
+	 * the connections to controllers returned in the Discovery Log Pages.
+	 * This is essential when invoking "connect-all" with --device to
+	 * reuse a persistent discovery controller (as is done for the udev
+	 * rules). In this case, we want host-traddr/host-iface to be
+	 * consistent with the discovery controller (c).
+	 */
+	if (!cfg.host_traddr)
+		cfg.host_traddr = (char *)nvme_ctrl_get_host_traddr(c);
+	if (!cfg.host_iface)
+		cfg.host_iface = (char *)nvme_ctrl_get_host_iface(c);
```

Comment:
I think this should be inside the `if (device & !force)` block. The the `force` option should be able to overwrite anything.

Reply:
Good point. Let me rework it.

Reply:
@igaw - Requested changes applied.